rcl_yaml_param_parser/src/parser.c

```diff
@@ -1105,7 +1106,30 @@ static void * get_value(
   {
     errno = 0;
     endptr = NULL;
-    dval = strtod(value, &endptr);
+    const char * iter_ptr = NULL;
+    if ((0 == strcmp(value, ".nan")) ||
+      (0 == strcmp(value, ".NaN")) ||
+      (0 == strcmp(value, ".NAN")) ||
+      (0 == strcmp(value, ".inf")) ||
+      (0 == strcmp(value, ".Inf")) ||
+      (0 == strcmp(value, ".INF")) ||
+      (0 == strcmp(value, "+.inf")) ||
+      (0 == strcmp(value, "+.Inf")) ||
+      (0 == strcmp(value, "+.INF")) ||
+      (0 == strcmp(value, "-.inf")) ||
+      (0 == strcmp(value, "-.Inf")) ||
+      (0 == strcmp(value, "-.INF")))
```

Comment:
Once we get https://github.com/ros2/rcutils/pull/280 in, we can change this code to use `rcutils_strcasecmp`.

Reply:
> Once we get [ros2/rcutils#280](https://github.com/ros2/rcutils/pull/280) in, we can change this code to use `rcutils_strcasecmp`.

@clalancette 
FYI (https://github.com/ros2/rcl/pull/741#discussion_r475394558)
The yaml spec only allows specific formats.

Besides, the existed build error doesn't seem to be caused by this PR.

Reply:
I checked it, the build error result from [#727](https://github.com/ros2/rcl/pull/727), the [rcutils](https://github.com/ros2/rcutils/pull/264/files) dependency has been merged, but build system still based on ros-rolling-rcutils: 1.1.0, which is the problem.

Anyway, this has no effect on this merge process.